generators/php/codegen/src/context/AbstractPhpGeneratorContext.ts

```diff
@@ -245,15 +381,33 @@ export abstract class AbstractPhpGeneratorContext<
         return declaration.shape.type === "enum";
     }
 
+    public isPrimitive(typeReference: TypeReference): boolean {
+        switch (typeReference.type) {
+            case "primitive": {
+                return true;
+            }
+            case "named": {
+                const declaration = this.getTypeDeclarationOrThrow(typeReference.typeId);
+                if (declaration.shape.type === "alias") {
+                    return this.isPrimitive(declaration.shape.aliasOf);
+                }
+                return false;
+            }
+            default: {
+                return false;
+            }
+        }
+    }
+
     public isDate(typeReference: TypeReference): boolean {
-        return this.isPrimitive({ typeReference, primitive: PrimitiveTypeV1.Date });
+        return this.isAliasOfPrimitive({ typeReference, primitive: PrimitiveTypeV1.Date });
     }
 
     public isDateTime(typeReference: TypeReference): boolean {
-        return this.isPrimitive({ typeReference, primitive: PrimitiveTypeV1.DateTime });
+        return this.isAliasOfPrimitive({ typeReference, primitive: PrimitiveTypeV1.DateTime });
     }
 
-    public isPrimitive({
+    public isAliasOfPrimitive({
```

Comment:
nit: I'd keep this naming as-is since it's implied aliases are handled.

Reply:
done